### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Category.Preorder

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mo271** [comment]
still todo:

- [ ] two simp lemmas give "Left-hand side does not simplify, when using the simp lemma on itself. This usually means that it will never apply." Should those be removed?
- [ ] probably not add `noncomputable` to `opHomOfLe`, but how to do that?
- [ ] two `aesop_cat` that used to be `tidy` are not working. Is there a way to make `aesop_cat` work on these?

**Comment by mattrobball** [comment]
I checked the two simp linter complaints at they don't solve themselves so I guess remove them. 

It appears `alias` inject noncomputability. [Another example](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/bugs.20and.20issues.20in.20the.20Data.2EList.2EInfix.20port/near/319649645) I just used the original declaration. 

I didn't mess around much with `aesop_cat` for those two.

**Comment by mo271** [comment]
Thanks for the help and interesting to hear that `alias` injects computability...

File: Mathlib/CategoryTheory/Category/Preorder.lean [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+-- so we provide them by hand:
+@[simp]
+theorem Equivalence.toOrderIso_apply (e : X ≌ Y) (x : X) : e.toOrderIso x = e.functor.obj x :=
+  rfl
+#align category_theory.equivalence.to_order_iso_apply CategoryTheory.Equivalence.toOrderIso_apply
+
+@[simp]
+theorem Equivalence.toOrderIso_symm_apply (e : X ≌ Y) (y : Y) :
+    e.toOrderIso.symm y = e.inverse.obj y :=
+  rfl
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Category/Preorder.lean:195]: Started a zulip discussion about this issue: https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/simps.20and.20simpNF.20linter

File: Mathlib/CategoryTheory/Category/Preorder.lean [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+/-- Extract the underlying inequality from a morphism in a preorder category.
+-/
+theorem leOfHom {x y : X} (h : x ⟶ y) : x ≤ y :=
+  h.down.down
+#align category_theory.le_of_hom CategoryTheory.leOfHom
+
+alias leOfHom ← _root_.Quiver.Hom.le
+#align quiver.hom.le Quiver.Hom.le
+
+-- porting note: linter gives: "Left-hand side does not simplify, when using the simp lemma on
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Category/Preorder.lean:93]: I've seen this linter claim nonsense before. Could you please try to prove
```
example {x y : X} (h : x ≤ y) : h.hom.le = h :=
  by simp
```
If it works: please restore the simp, add a `nolint simpNF` and adjust the porting note to indicate the the linter is wrong.

Comment (mo271) [reply Mathlib/CategoryTheory/Category/Preorder.lean:93]: Your example seems to work, but simp fails on
```
example {x y : X} (h : x ⟶ y) : h.le.hom = h :=
  by simp
````

Comment (mo271) [reply Mathlib/CategoryTheory/Category/Preorder.lean]: pushing the nolint for the first case...

Comment (mo271) [reply Mathlib/CategoryTheory/Category/Preorder.lean]: So, for one case the linter seems to be right and not for the other?!

File: Mathlib/CategoryTheory/Category/Preorder.lean [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+theorem Equivalence.toOrderIso_apply (e : X ≌ Y) (x : X) : e.toOrderIso x = e.functor.obj x :=
+  rfl
+#align category_theory.equivalence.to_order_iso_apply CategoryTheory.Equivalence.toOrderIso_apply
+
+@[simp]
+theorem Equivalence.toOrderIso_symm_apply (e : X ≌ Y) (y : Y) :
+    e.toOrderIso.symm y = e.inverse.obj y :=
+  rfl
+#align category_theory.equivalence.to_order_iso_symm_apply
+    CategoryTheory.Equivalence.toOrderIso_symm_apply
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Category/Preorder.lean:196]:
```suggestion
#align category_theory.equivalence.to_order_iso_symm_apply CategoryTheory.Equivalence.toOrderIso_symm_apply
```

File: Mathlib/CategoryTheory/Category/Preorder.lean (hunk 2 above)
Comment (jcommelin) [reply Mathlib/CategoryTheory/Category/Preorder.lean]: apparently

**Comment by jcommelin** [comment]
I'm going to merge this. The issue with simps has been reported. We can revisit this file after it is fixed.

bors merge

**Pull request closed without merging**